modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/kink_collars.dm

```diff
@@ -87,33 +133,50 @@
 	desc = "A tight collar. It looks like it has some kind of lock."
 	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_clothing/lewd_neck.dmi'
 	worn_icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_neck.dmi'
-	icon_state = "lock_collar_cyan"
-	inhand_icon_state = "lock_collar_cyan"
+	icon_state = "lock_collar"
+	inhand_icon_state = "lock_collar"
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/small/kink_collar/locked
 	treat_path = /obj/item/key/kink_collar
 	var/lock = FALSE
 	var/broke = FALSE
 	var/key_id = null //Adding unique id to collar
-	unique_reskin = list("Cyan" = "lock_collar_cyan",
-						"Yellow" = "lock_collar_yellow",
-						"Green" = "lock_collar_green",
-						"Red" = "lock_collar_red",
-						"Latex" = "lock_collar_latex",
-						"Orange" = "lock_collar_orange",
-						"White" = "lock_collar_white",
-						"Purple" = "lock_collar_purple",
-						"Black" = "lock_collar_black",
-						"Black-teal" = "lock_collar_tealblack",
-						"Spike" = "lock_collar_spike")
-
-//spawn thing in collar
+	var/static/list/lock_collar_designs
+
+//create radial menu
+/obj/item/clothing/neck/kink_collar/locked/proc/populate_lock_collar_designs()
+	lock_collar_designs = list(
+		"cyan" = image (icon = src.icon, icon_state = "lock_collar_cyan"),
+		"yellow" = image(icon = src.icon, icon_state = "lock_collar_yellow"),
+		"green" = image(icon = src.icon, icon_state = "lock_collar_green"),
+		"red" = image(icon = src.icon, icon_state = "lock_collar_red"),
+		"latex" = image(icon = src.icon, icon_state = "lock_collar_latex"),
+		"orange" = image(icon = src.icon, icon_state = "lock_collar_orange"),
+		"white" = image(icon = src.icon, icon_state = "lock_collar_white"),
+		"purple" = image(icon = src.icon, icon_state = "lock_collar_purple"),
+		"black" = image(icon = src.icon, icon_state = "lock_collar_black"),
+		"tealblack" = image(icon = src.icon, icon_state = "lock_collar_tealblack"),
+		"spike" = image(icon = src.icon, icon_state = "lock_collar_spike"))
 
 //reskin code
 
-/obj/item/clothing/neck/kink_collar/locked/AltClick(mob/user)
+/obj/item/clothing/neck/kink_collar/locked/AltClick(mob/user, obj/item/I)
```

Comment:
Unnecessarily duplicated proc, modify proc on parent to account for this instead of repeating it